workers/src/bin/dvn.rs

```diff
@@ -0,0 +1,118 @@
+                    }
+                    Ok(inner_log) => {
+                        debug!("PacketSent event found and decoded");
+                        dvn_data.packet_received(inner_log.data().clone());
+                    },
+                }
+            }
+            Some(log) = sendlib_stream.next() => {
+                match log.log_decode::<DVNFeePaid>() {
+                    Err(e) => {
+                        error!("Received a `DVNFeePaid` event but failed to decode it: {:?}", e);
+                    }
+                    Ok(inner_log) => {
+                        if dvn_data.packet.is_some() {
+
+                            info!("DVNFeePaid event found and decoded.");
+                            let required_dvns = inner_log.inner.requiredDVNs.clone();
+                            let own_dvn_addr = dvn_data.config.dvn_addr.parse::<Address>()?;
+
+                            if required_dvns.contains(&own_dvn_addr) {
+                                debug!("Found DVN in required DVNs.");
+
+                                // Query how many confirmations are required.
+                                let eid = U256::from(dvn_data.config.network_eid);
+                                let required_confirmations = query_confirmations(&receivelib_contract, eid).await?;
+
+                                // Prepare the header hash.
+                                let header_hash = dvn_data.get_header_hash();
+                                // Prepate the payload hash.
+                                let message_hash = dvn_data.get_message_hash();
+
+                                // Check if the info from the payload could have been extracted.
+                                if let (Some(header), Some(payload)) = (header_hash, message_hash) {
+                                    let already_verified = query_already_verified(
+                                        &receivelib_contract,
+                                        own_dvn_addr,
+                                        header.as_ref(),
+                                        payload.as_ref(),
+                                        required_confirmations,
+                                    )
+                                    .await?;
+
+                                    if already_verified {
+                                        debug!("Packet already verified.");
+                                    } else {
+                                        dvn_data.verifying();
+                                        debug!("Packet NOT verified. Calling verification.");
+
+                                        // FIXME: logic for NFFL verification
+
+                                        verify(
+                                            &receivelib_contract,
+                                            // SAFETY: if-let above fails before this.
+                                            &dvn_data.get_header().unwrap().to_slice(),
```

Comment:
but we already have the header then? Actually, we don't. This is unsafe, use nullish-coalescing